# HG changeset patch
# User andrew
# Date 1421873872 0
#      Wed Jan 21 20:57:52 2015 +0000
# Node ID 31845b0edbd16232f26c266f1f79729fb83a6975
# Parent  41b13b19b58f4945f18828e8a7ecf35c3a12a596
OPENJDK6-52: Application of 6786276 introduces compatibility issue
Summary: Add back ISO-3166 code CS and adjust LocaleTest to account for the extra country.
Reviewed-by: omajid

diff -r 41b13b19b58f -r 31845b0edbd1 src/share/classes/java/util/LocaleISOData.java
--- openjdk/jdk/src/share/classes/java/util/LocaleISOData.java	Wed Jan 21 20:27:18 2015 +0000
+++ openjdk/jdk/src/share/classes/java/util/LocaleISOData.java	Wed Jan 21 20:57:52 2015 +0000
@@ -275,7 +275,7 @@
         + "CN" + "CHN"  // China, People's Republic of
         + "CO" + "COL"  // Colombia, Republic of
         + "CR" + "CRI"  // Costa Rica, Republic of
-//      + "CS" + "SCG"  // Serbia and Montenegro
+        + "CS" + "SCG"  // Serbia and Montenegro
         + "CU" + "CUB"  // Cuba, Republic of
         + "CV" + "CPV"  // Cape Verde, Republic of
         + "CW" + "CUW"  // Cura\u00e7ao
diff -r 41b13b19b58f -r 31845b0edbd1 test/java/util/Locale/Bug6786276.java
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ openjdk/jdk/test/java/util/Locale/Bug6786276.java	Wed Jan 21 20:57:52 2015 +0000
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2015 Red Hat, Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+ * @test
+ * @bug 6786276
+ * @summary Verify that the ISO-3166 code for Serbia & Montenegro is still present
+ */
+
+import java.util.Arrays;
+import java.util.Locale;
+
+public class Bug6786276
+{
+    public static void main(String[] args)
+    {
+	String[] localeCodes = Locale.getISOCountries();
+	System.err.println("Locale codes: " + Arrays.toString(localeCodes));
+	int serbiaMontenegro = Arrays.binarySearch(localeCodes, "CS");
+	if (serbiaMontenegro >= 0) {
+	    System.out.println("Serbia & Montenegro ISO code present, index "
+			       + serbiaMontenegro);
+	} else {
+	    System.out.println("Serbia & Montenegro ISO code not present");
+	    throw new RuntimeException("Serbia & Montenegro ISO code not present.");
+	}
+    }
+}
